### PR TITLE
feat: effective attendance computation for all subjects + arrivals/departures

### PR DESCRIPTION
**Pull request by kouloumos**
## Summary

Addresses items `#1`, `#2`, and `#3` from schemalabz/opencouncil#173 (Phase 2 follow-ups).

### Problem

Per-subject attendance was computed per-PDF in isolation — each subject only saw its own PDF's attendance data, discussion order, and name forms. This caused:
- **Name mismatches**: en-dashes vs spaces (`ΣΤΡΑΚΑΝΤΟΥΝΑ –ΣΦΑΚΑΚΗ` vs `ΣΤΡΑΚΑΝΤΟΥΝΑ ΣΦΑΚΑΚΗ`), abbreviated vs full names (`Κ. Αγγελής` vs `ΚΩΝΣΤΑΝΤΙΝΟΣ ΑΓΓΕΛΗΣ`) silently prevented attendance changes from being applied
- **Missing attendance for subjects without decisions**: no PDF = no attendance computation
- **Inconsistent votes**: FOR votes for unanimous decisions were inferred from per-PDF effective attendance, which often fell back to the initial roll call instead of accounting for mid-meeting arrivals/departures
- **Duplicate/conflicting changes**: same attendance change extracted from 20+ PDFs with different name variants and timing values

### Solution

**Single meeting-level computation path** — attendance changes and discussion order are aggregated from all PDFs once, names are resolved to canonical forms, and effective attendance is computed for every subject (regular + out-of-agenda) using the complete discussion order. Vote inference uses this corrected attendance.

Key changes:
- `resolveAndDeduplicateAttendanceChanges`: resolves names via `nameToPersonId`, groups by person+type+agendaItem, picks timing by majority vote across PDFs
- `buildCompleteDiscussionOrder`: merges the PDF's partial explicit order with the full subject list (OA first, then regular in numerical order)
- `computeAllSubjectAttendance`: computes effective attendance for every subject including OA (via `outOfAgendaIndex`)
- Pipeline Phase 3→4 restructure: attendance computed once at meeting level, vote inference uses that attendance directly. The old per-PDF `processRawExtraction` → overwrite pattern is eliminated
- `normalizeGreekName`: handles Unicode dashes; extraction prompt clarifies "πριν" → `timing: "during"` and adds first-initial abbreviation matching

Companion PR: schemalabz/opencouncil#XXX

<h3>Greptile Summary</h3>

This PR replaces per-PDF isolated attendance computation with a single meeting-level computation path. Attendance changes from all PDFs are aggregated and deduplicated once, names are resolved to canonical forms, and effective attendance is computed for every subject — including those without decision PDFs — using the complete discussion order.

- **`meetingAttendance.ts`** (new): `resolveAndDeduplicateAttendanceChanges` deduplicates attendance changes across PDFs using majority-vote timing and name canonicalization; `buildCompleteDiscussionOrder` merges the PDF's partial explicit order with the full subject list; `computeAllSubjectAttendance` computes per-subject attendance for the entire meeting.
- **`extractionPipeline.ts`**: Phase 3→4 restructure — attendance is computed once at meeting level, vote inference (FOR votes) uses the resolved per-subject attendance directly instead of falling back to the initial roll call; `nonDecisionSubjectAttendance` is exposed in the result for subjects without linked PDFs.
- **`effectiveAttendance.ts`**: Names in `present`/`absent` sets are keyed by `normalizeGreekName` (handling Unicode dashes, diacritics, case) so that minor name variants in attendance changes match the canonical initial-list form.

<h3>Confidence Score: 5/5</h3>

Safe to merge — no incorrect data paths identified; the meeting-level attendance computation is correct and well-tested.

All three core functions have thorough unit tests including real-world edge cases (Zografou ordering, en-dash normalization, name abbreviation). The tie-breaking logic for timing disagreements across PDFs is deterministic and correctly prefers 'during' over 'after' on ties. Vote inference in Phase 4 is semantically equivalent to the retired inferForVotes helper. The only structural limitation is the 'longest wins' heuristic for discussion-order selection, which is mitigated by buildCompleteDiscussionOrder appending any missing subjects.

No files require special attention. extractionPipeline.ts has a minor heuristic limitation in bestDiscussionOrder selection worth revisiting if ordering disagreements across PDFs occur in practice.

<h3>Important Files Changed</h3>

| Filename | Overview |
|----------|----------|
| src/tasks/utils/meetingAttendance.ts | New file implementing meeting-level attendance aggregation. Tie-breaking logic (prefer 'during' over 'after' on equal counts) is correctly implemented. Deduplication key correctly includes name+type+agendaItem. |
| src/tasks/utils/extractionPipeline.ts | Pipeline restructured from per-PDF to meeting-level attendance. Vote inference logic replicated inline correctly matches the semantics of the retired inferForVotes helper. |
| src/tasks/utils/effectiveAttendance.ts | Name normalization via canonicalName map correctly preserves first-seen (initial-list) form as canonical. Early-return on empty attendanceChanges bypasses normalization but returns raw initial-list names, which are valid keys in nameToPersonId. |
| src/tasks/utils/meetingAttendance.test.ts | Good test coverage of deduplication, majority-vote timing, tie-breaking, complete discussion order construction, and the real-world Zografou case. |
| src/tasks/pollDecisions.ts | OA counter assignment is a reasonable fallback, correctly overridden per-subject by oaIndexFromExtraction when a PDF is available. |
| src/tasks/utils/decisionPdfExtraction.ts | Unicode dash normalization range is correct and intentionally excludes ASCII hyphen-minus to preserve compound surnames. |
| src/types.ts | nonAgendaReason added as optional string | null. nonDecisionSubjectAttendance correctly typed as optional array. |

</details>

<h3>Flowchart</h3>

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
    A[extractDecisionsFromPdfs] --> B[Phase 1: Extract all PDFs in batches]
    B --> C[Phase 2: Meeting-level name matching\ntokenSort + LLM fallback → nameToPersonId]
    C --> D[resolveAndDeduplicateAttendanceChanges\nname canonicalization + majority-vote timing]
    C --> E[Select bestDiscussionOrder\nlongest from any PDF]
    D --> F[MeetingAttendanceData]
    E --> F
    F --> G[Phase 3: computeAllSubjectAttendance\nbuildCompleteDiscussionOrder]
    G --> H[computeEffectiveAttendance per subject\nnormalized name keys → personId lookup]
    H --> I[attendanceBySubject Map\n+ nonDecisionSubjectAttendance]
    I --> J[Phase 4: Build decision results\nexplicit votes + inferred FOR votes]
    J --> K[ExtractionPipelineResult]
```

<details><summary>Prompt To Fix All With AI</summary>

`````markdown
Fix the following 1 code review issue. Work through them one at a time, proposing concise fixes.

---

### Issue 1 of 1
src/tasks/utils/extractionPipeline.ts:187-193
When two or more PDFs each have a `discussionOrder` of the same maximum length, this picks whichever appears first in the `extractions` array rather than the most informative one. More importantly, if PDF-A reports `[1, OA1, 9]` and PDF-B reports `[OA2, 2, 3]` (different items, same length), the selected order omits the items unique to the non-selected PDF entirely. `buildCompleteDiscussionOrder` will still append the missing subjects, but they'll be appended in sorted order rather than at their true meeting position. Merging unique items from all PDFs' discussion orders before passing to `buildCompleteDiscussionOrder` would be more robust.

```suggestion
        // Use the longest discussion order from any PDF (most complete).
        // When two PDFs have the same length but different items, the first
        // one wins; consider merging unique items across all PDFs if this
        // becomes a problem in practice.
        let bestDiscussionOrder: AgendaItemRef[] | null = null;
        for (const { raw } of extractions) {
            if (raw.discussionOrder && (bestDiscussionOrder === null || raw.discussionOrder.length > bestDiscussionOrder.length)) {
                bestDiscussionOrder = raw.discussionOrder;
            }
        }
```

`````

</details>

<sub>Reviews (4): Last reviewed commit: ["fix: assign OA indices from extraction s..."](https://github.com/schemalabz/opencouncil-tasks/commit/dcb4287888bc38d0117d9b0657e5f5b46ba2613f) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=32536514)</sub>

### CODE REVIEW

**Comment by kouloumos** [comment]
I [force-pushed](https://github.com/schemalabz/opencouncil-tasks/compare/20795d4605d7ab1f1eac300d05f58d7fe8b71fea..721507c617f655c0b95ef1decf5a90e44b673351) to address bot review comment ([1](https://github.com/schemalabz/opencouncil-tasks/pull/40#discussion_r3255628040)):

- Stabilized tie-breaking in timing majority vote: prefer "during" over "after" on equal counts (more conservative — assumes person was already absent)

File: src/tasks/utils/meetingAttendance.ts [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+    for (const { change, timingVotes } of groups.values()) {
+        // Pick timing with most votes; on tie, prefer specified over null
+        let bestTiming = change.timing;
+        let bestCount = 0;
+        for (const [timing, count] of timingVotes) {
+            if (count > bestCount || (count === bestCount && timing !== 'null')) {
+                bestTiming = timing === 'null' ? null : timing as 'during' | 'after';
+                bestCount = count;
+            }
+        }
```
[the remark this answers is not in the document]
Comment (kouloumos) [reply src/tasks/utils/meetingAttendance.ts:87]: Fixed in [force-push](https://github.com/schemalabz/opencouncil-tasks/compare/20795d4605d7ab1f1eac300d05f58d7fe8b71fea..721507c617f655c0b95ef1decf5a90e44b673351) — stabilized tie-breaking to prefer "during" over "after" on equal counts

File: src/tasks/pollDecisions.ts [hunk 2]
```diff
@@ -596,8 +596,23 @@ export const pollDecisions: Task<PollDecisionsRequest, PollDecisionsResult> = as
...
+        // (including those without decisions) using the complete discussion order.
+        let oaCounter = 0;
+        const allMeetingSubjects = request.subjects.map(s => {
+            const isOA = s.nonAgendaReason === 'outOfAgenda';
+            if (isOA) oaCounter++;
+            return {
+                subjectId: s.subjectId,
+                agendaItemIndex: s.agendaItemIndex,
+                outOfAgendaIndex: isOA ? oaCounter : null,
+            };
```
[the remark this answers is not in the document]
Comment (kouloumos) [reply src/tasks/pollDecisions.ts:610]: Not a bug today — OA subjects currently arrive with `agendaItemIndex: null`. In the future OA items may get their own `agendaItemIndex` (they have their own numbering in practice), so nulling it at the call site would discard useful data. If both fields are ever set simultaneously, the priority logic in `computeAllSubjectAttendance` should be updated instead.
